utils/appliance/__init__.py

```diff
@@ -2215,19 +2204,13 @@ def push(self, obj):
 
         logger.info("Pushed appliance {} on stack (was {} before) ".format(
             obj.address, getattr(was_before, 'address', 'empty')))
-        if obj.browser_steal:
```

Comment:
i see, i feared something like this might happen, the browser steal code-path is to my understanding the problem source for appliance pop issues